src/ejabberd_cowboy.erl

```diff
@@ -218,3 +223,12 @@ store_trails(Routes) ->
               ?LOG_WARNING(#{what => store_trails_failed,
                              class => Class, reason => Reason, stacktrace => Stacktrace})
     end.
+
+-spec handler_instumentation(module()) -> [mongoose_instrument:spec()].
+handler_instumentation(Module) ->
+    case erlang:function_exported(Module, http_handler_instrumentation, 0) of
+        true ->
+            Module:http_handler_instrumentation();
```

Comment:
This should be delegated to something like `mongoose_http_handler:instrumentation(Module)`